blockchain/message_pool/src/msgpool.rs

```diff
@@ -214,10 +215,10 @@ where
             loop {
                 if let Some(ts) = subscriber.next().await {
                     head_change(
-                        api.clone(),
-                        bls_sig_cache.clone(),
-                        pending.clone(),
-                        cur_tipset.clone(),
+                        api.as_ref(),
```

Comment:
just so you know, you don't have to use `as_ref()` explicitly, can just put a `&` behind the variables if it's easier in future